api/options.go

```diff
@@ -0,0 +1,104 @@
+package api
+
+import "go.uber.org/zap"
+
+// ServerOption is a functional option to configure an API server.
+type ServerOption func(*api)
+
+// ServerWithAlerts sets the alerts manager for the API server.
+func ServerWithAlerts(al Alerts) ServerOption {
+	return func(a *api) {
+		a.alerts = al
+	}
+}
+
+// ServerWithWebHooks sets the webhooks manager for the API server.
+func ServerWithWebHooks(w WebHooks) ServerOption {
```

Comment:
`nit` we treat webhook like filename essentially and write it as `Webhook` (which I think is more correct)

Reply:
It's `WebHooks` everywhere in `hostd`, we can handle it in a fu.